spring-kafka/src/main/java/org/springframework/kafka/core/TransactionSuffixManager.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2016-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.kafka.core;
+
+/**
+ * The strategy for managing transactional producer suffixes.
```

Comment:
If it's a real strategy, we must call it such. But are we providing a way to customize a strategy? Are we envisioning a situation where users provide a custom implementation in which they implement a different strategy? (i.e., non-incrementing integers for suffix, but rather a random string, for e.g.)

Reply:
Yes, it's a real strategy. The main reason why I created this issue is to provide the ability to manage `txIds` using external storage, for example, because in my environment, it's a better strategy than everything else. So maybe it's better to call it `TransactionSuffixStrategy`/`DefaultTransactionSuffixStrategy`?

Reply:
Strategy needs a strategy method. Here has three method `setMaxCache`,  `getSuffix`, `releaseSuffix`, not like strategy

Reply:
> Yes, it's a real strategy. The main reason why I created this issue is to provide the ability to manage `txIds` using external storage, for example, because in my environment, it's a better strategy than everything else. So maybe it's better to call it `TransactionSuffixStrategy`/`DefaultTransactionSuffixStrategy`?

`TransactionIdSuffixStrategy`/`DefaultTransactionIdSuffixStrategy`?